scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -64,17 +64,21 @@ interface ScanResultsStorage {
 
         override fun read(id: Identifier) =
                 storage.read(id).also {
-                    ++stats.numReads
-                    if (it.results.isNotEmpty()) {
-                        ++stats.numHits
+                    synchronized(stats) {
```

Comment:
The stats variable could be turned from `var` to `val``to make it obvious that we synchronize always with the same object.

Reply:
It currently needs to be a `var` because `FileCounterTest` resets / reassigns it in the end. I'll think about introducing a `reset()` function instead.

Reply:
yeah, the reset function would also nicely fit into the AccessStatistics class imo.